src/azure-cli/azure/cli/command_modules/role/util.py

```diff
@@ -0,0 +1,78 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+
+def _get_current_user_object_id(graph_client):
+    # Calling this API with a service principal raises GraphError:
+    #   /me request is only valid with delegated authentication flow.
+    current_user = graph_client.signed_in_user_get()
+    return current_user['id']
+
+
+def _get_object_id_by_spn(graph_client, spn):
+    accounts = list(graph_client.service_principal_list(filter="servicePrincipalNames/any(c:c eq '{}')".format(spn)))
+    # 2+ matches should never happen, so we only check 'no match' here
+    if not accounts:
+        logger.warning("Unable to find service principal with spn '%s'", spn)
+        return None
+    return accounts[0]['id']
+
+
+def _get_object_id_by_upn(graph_client, upn):
+    accounts = list(graph_client.user_list(filter="userPrincipalName eq '{}'".format(upn)))
+    # 2+ matches should never happen, so we only check 'no match' here
+    if not accounts:
+        logger.warning("Unable to find user with upn '%s'", upn)
+        return None
+    return accounts[0]['id']
+
+
+def _get_object_id_from_subscription(graph_client, subscription):
+    if subscription['user']['type'] == 'user':
+        return _get_object_id_by_upn(graph_client, subscription['user']['name'])
+    if subscription['user']['type'] == 'servicePrincipal':
+        return _get_object_id_by_spn(graph_client, subscription['user']['name'])
+    return None
+
+
+def _get_object_id(graph_client, spn=None, upn=None):
```

Comment:
There is no need for `_get_object_id` to support `subscription` argument. `_get_object_id_from_subscription` can be directly called by `get_current_identity_object_id`.